File: Homeworks/C# Part 2/06.StringsAndTextProcessing/15.ReplaceTags/ReplaceTags.cs
using System;
using System.Text;

class ReplaceTags
{
    static void Main()
    {
        var text = Console.ReadLine();
        var result = new StringBuilder(text.Length);
        int searchIndex = 0;
        while (true)
        {
            int appendIndex = text.IndexOf("<a href=", searchIndex);
            if (appendIndex < 0)
            {
                result.Append(text.Substring(searchIndex));
                break;
            }
            result.Append(text.Substring(searchIndex, appendIndex - searchIndex));
            int linkBeginIndex = text.IndexOf("\"", appendIndex + 7);
            int linkEndIndex = text.IndexOf("\"", linkBeginIndex + 1);
            int nameBeginIndex = text.IndexOf(">", linkEndIndex + 1);
            int nameEndIndex = text.IndexOf("<", nameBeginIndex + 1);
            string link = text.Substring(linkBeginIndex + 1, linkEndIndex - linkBeginIndex - 1);
            string name = text.Substring(nameBeginIndex + 1, nameEndIndex - nameBeginIndex - 1);
            result.Append(string.Format("[{0}]({1})", name, link));
            searchIndex = nameEndIndex + 4;
        }
        Console.WriteLine(result);
    }
}

File: Homeworks/JS Fundamentals/11. Strings/08. Replace tags/replace-tags.js
function solve(args) { 
    var result = '';
        searchIndex = 0;
    while (true)
    {
        var appendIndex = args[0].indexOf("<a href=", searchIndex);
        if (appendIndex < 0)
        {
            result += args[0].substr(searchIndex);
            break;
        }
        result += args[0].substr(searchIndex, appendIndex - searchIndex);
        var linkBeginIndex = args[0].indexOf("\"", appendIndex + 7);
        var linkEndIndex = args[0].indexOf("\"", linkBeginIndex + 1);
        var nameBeginIndex = args[0].indexOf(">", linkEndIndex + 1);
        var nameEndIndex = args[0].indexOf("<", nameBeginIndex + 1);
        var link = args[0].substr(linkBeginIndex + 1, linkEndIndex - linkBeginIndex - 1);
        var name = args[0].substr(nameBeginIndex + 1, nameEndIndex - nameBeginIndex - 1);
        result += '[' + name + '](' + link + ')';
        searchIndex = nameEndIndex + 4;
    }
    console.log(result);
}